### DISCUSSION TITLE
Variable theme - enable the potential to use more css variables

### DISCUSSION THREAD

**Discussion by imaliouris** [NG-ZORRO/ng-zorro-antd#8694] [Ideas]
Currently in the variable theme we can only configure colors as css variables and I was wondering if we could refactor some of the code in order to be able to set i.e. `@select-multiple-item-height` as a css variable.

In current implementation this can't happen because if we set 
```less
@select-multiple-item-height: ~"calc(@{input-height-base}-@{input-padding-vertical-base}*2)";
``` 
less compiler will throw an error because in `select/style/multiple.less` we have a calculation at line 78 
```less
line-height: @select-multiple-item-height - @select-multiple-item-border-width * 2;
```

My proposal is to refactor every calculation in component styles to use css `calc()` instead of less calculations.
So in the previous example that would result to 
```less
line-height: ~"calc(@{select-multiple-item-height}-@{select-multiple-item-border-width}*2)";
```

**Comment by lppedd** [comment]
Adding myself to this request.

The variable theme has proven to be fantastic, but LESS functions need to be replaced with standard CSS functionalities, or dropped, as they can't be easily extended.

**Reply by lppedd** [reply]
@imaliouris better to move this to an issue btw.
